otcextensions/sdk/cloud/dds.py

```diff
@@ -0,0 +1,186 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+from openstack import exceptions
+
+
+class DdsMixin:
+    def create_dds_instance(self, name, **kwargs):
+        """Create DDS instance with all the checks
+
+        :param str name: dict(required=True, type=str),
+        :param str datastore_type: dict(type=str, default='DDS-Community'),
+        :param str datastore_version: dict(type=str),
+        :param str datastore_storage_engine:
+            dict(type=str, default='wiredTiger'),
+        :param str region:
+            dict(type=str, default='eu-de'),
+        :param str availability_zone: dict(type=str),
+        :param str router: dict(type=str),
+        :param str network: dict(type=str),
+        :param str security_group: dict(type=str),
+        :param str password: dict(type=str, no_log=True),
+        :param str disk_encryption_id: dict(type=str),
+        :param str mode: choices=['Sharding', 'ReplicaSet']
+        :param str flavors: dict(required=True, type=list, elements=dict),
+        :param str backup_timeframe: dict(type=str),
+        :param str backup_keepdays: dict(type=str),
+        :param str ssl_option: dict(type=str),
+
+        :returns: The results of instance creation
+        :rtype: :class:`~otcextensions.sdk.dds.v3.instance.Instance`
+        """
+
+        datastore_type = kwargs.get('datastore_type')
+        datastore_version = kwargs.get('datastore_version')
+        datastore_storage_engine = kwargs.get('datastore_storage_engine')
+        region = kwargs.get('region')
+        availability_zone = kwargs.get('availability_zone')
+        router = kwargs.get('router')
+        network = kwargs.get('network')
+        security_group = kwargs.get('security_group')
+        password = kwargs.get('password')
+        disk_encryption_id = kwargs.get('disk_encryption_id')
+        mode = kwargs.get('mode')
+        flavors = kwargs.get('flavors')
+        backup_timeframe = kwargs.get('backup_timeframe')
+        backup_keepdays = kwargs.get('backup_keepdays')
+        ssl_option = kwargs.get('ssl_option')
+
+        attrs = {}
+        attrs['name'] = name
+
+        if datastore_type:
+            datastore = {
+                'type': datastore_type,
+                'version': datastore_version,
+                'storage_engine': datastore_storage_engine
+            }
+            attrs['datastore'] = datastore
+        elif datastore_version or datastore_storage_engine:
+            raise exceptions.SDKException(
+                '`datastore_type` and `datastore_version`'
+                ' and `datastore_storage_engine` must be passed together'
+            )
+
+        if region:
```

Comment:
mandatory

Reply:
fixed

Reply:
well, it is actually easier just to include mandatory params in the function definition